### ISSUE TITLE
Increased VRAM usage with ControlNet when torch.backends.cudnn.benchmark=True

### ISSUE THREAD

**Issue by d1g1t** [huggingface/diffusers#2763]
### Describe the bug

Not entirely sure if this is a bug but setting `torch.backends.cudnn.benchmark=True` as per https://huggingface.co/docs/diffusers/optimization/fp16 significantly increases vram usage when used with `StableDiffusionControlNetPipeline`

With the example script I get the output
`2,746,436,608, 2,938,109,952` without `torch.backends.cudnn.benchmark=True` and
`8,197,157,888, 8,600,420,352` with `torch.backends.cudnn.benchmark=True`

If this is intended, perhaps the optimizaton page can mention which pipelines are compatible/incompatible with the optimizations.
Similarly `StableDiffusionControlNetPipeline` does not appear to support `enable_vae_tiling()`

### Reproduction
```python
from diffusers import StableDiffusionControlNetPipeline, ControlNetModel, UniPCMultistepScheduler
from diffusers.utils import load_image
import numpy as np
import torch
import cv2
from PIL import Image

torch.backends.cudnn.benchmark = True

# download an image
image = load_image(
    "https://hf.co/datasets/huggingface/documentation-images/resolve/main/diffusers/input_image_vermeer.png"
)
image = np.array(image)
# get canny image
image = cv2.Canny(image, 100, 200)
image = image[:, :, None]
image = np.concatenate([image, image, image], axis=2)
canny_image = Image.fromarray(image)
# load control net and stable diffusion v1-5
controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny", torch_dtype=torch.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16
)
# speed up diffusion process with faster scheduler and memory optimization
pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
# remove following line if xformers is not installed
pipe.enable_xformers_memory_efficient_attention()
pipe.enable_model_cpu_offload()
# generate image
generator = torch.manual_seed(0)
image = pipe(
    "futuristic-looking woman", num_inference_steps=20, generator=generator, image=canny_image
).images[0]
image.save('controlnet_test.png')

print(f"{torch.cuda.max_memory_allocated():,}, {torch.cuda.max_memory_reserved():,}")
```

### System Info

- huggingface_hub version: 0.12.1
- Platform: Linux-5.15.0-67-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /home/d1g1t/.cache/huggingface/token
- Has saved token ?: True
- Who am I ?: d1g1t
- Configured git credential helpers: 
- FastAI: N/A
- Tensorflow: N/A
- Torch: 1.13.1
- Jinja2: 2.11.3
- Graphviz: N/A
- Pydot: N/A
- Pillow: 9.4.0
- hf_transfer: N/A
- ENDPOINT: https://huggingface.co
- HUGGINGFACE_HUB_CACHE: /home/d1g1t/.cache/huggingface/hub
- HUGGINGFACE_ASSETS_CACHE: /home/d1g1t/.cache/huggingface/assets
- HF_HUB_OFFLINE: False
- HF_TOKEN_PATH: /home/d1g1t/.cache/huggingface/token
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False

**Comment by patrickvonplaten** [comment]
Hmm that's interesting! I don't think I'll have time to dig deeper here, but also interested in case someone figures out what's going on here

**Comment by Stax124** [comment]
> Hmm that's interesting! I don't think I'll have time to dig deeper here, but also interested in case someone figures out what's going on here

We had the same issue on Volta - that benchmark will run only the first time you infer with the model and it will occupy the same amount of VRAM as the model alongside it. I spent multiple days tracing this mem leak. It would be nice if it was taken out of the docs so that more devs do not need to go through this process.

TLDR: 2x VRAM usage on the first run for the same or even worse results.

**Comment by patrickvonplaten** [comment]
hmm @williamberman @pcuenca @huggingface/transformers @patil-suraj @sayakpaul what do you think here, should me remove this from the docs?

**Comment by pcuenca** [comment]
I'd say we remove it and let advanced users investigate on their own. I feel this is more or less unnecessary after the built-in optimizations that come from PyTorch 2.

**Comment by Stax124** [comment]
@d1g1t can you close this as it was merged? thanks
